### PR TITLE
Port fix for mismatched NuGet versions to release/2.1.1xx.

### PR DESCRIPTION
**Pull request by peterhuene**
This commit ports the fix for the mismatched NuGet version numbers used by the
SDK.

See https://github.com/dotnet/sdk/pull/1831 for the change into master.

### CODE REVIEW

**Comment by peterhuene** [comment]
Do not merge without approval.

**Comment by nguerrera** [comment]
**Customer scenario**

Perf. Using VS 15.6 can result in extra copies of NuGet DLLs being loaded SxS.

Furthermore, this change was required to produce a source build because there cannot be more than version of NuGet.


**Bugs this fixes:** 

[VSO 574971](https://devdiv.visualstudio.com/DevDiv/_queries?id=574971&triage=true&_a=edit)

**Workarounds, if any**

Accept the perf hit.

Source build can apply a patch, but this keeps a discrepancy betweeen source and non-source-builds and adds debt. By putting this low risk fix into the next servicing release, we eliminate that debt sooner.

**Risk**

Low

**Performance impact**

Can reduce number of assemblies loaded in devenv.exe.

**Is this a regression from a previous update?**

Yes. The process of adjusting the repo to repo toolset and integrating a Version.props with separate fields for the 3 separate nuget packages, combined with incorrect merges from branch to branch to revert the version of NuGet referenced by the build task (and shipped alongside it) to a previous RTM version.

**Root cause analysis:**

Source build needs to be continuously updated (work is in progress on this).

Version.props should never duplicate versions, but instead reference something that needs to have the same version by name. This is what the patch here does.

**How was the bug found?**

Preparing source build.

**Review by livarcocc — APPROVED** [review]

**Comment by livarcocc** [comment]
@MattGertz we need this fix. I am approving and will chat with you tomorrow. Will take out if you disagree. But I want to cook up the VS insertion build tonight.

**Comment by MattGertz** [comment]
No disagreement.

**Pull request merged**
